### DISCUSSION TITLE
Managing NULL-ID apps

### DISCUSSION THREAD

**Discussion by AlessandroSerra** [nikitabobko/AeroSpace#772] [questions-and-answers] [answered]
I'm using aerospace with some app-dependent configuration such as which apps should have a floating leyout. I'm having some issues because some programs i use don't have a native apple-assigned ID, for example a visualization software named "pymol" is represented as:

12951 | NULL-APP-BUNDLE-ID     | python

or another program named VMD as:

12983 | NULL-APP-BUNDLE-ID     | VMD

I'd like aerospace to never manage those programs but I'm not sure how, because for other program apple is able to automatically assign an ID so I can just:

[[on-window-detected]]
if.app-id = 'com.spotify.client'
run = 'layout floating'

Anyone has any clue on what I could do?

**Comment by dirichy** [comment]
Met same question. May be I think we should add `if.null-app-id` condition.

**Comment by nikitabobko** [comment, marked as answer]
Since 0.21.0-Beta, you can use the new [test command](https://nikitabobko.github.io/AeroSpace/commands#test) and slightly updated syntax for on-window-detected ( [docs](https://nikitabobko.github.io/AeroSpace/guide#on-window-detected-callback) ) to create more sophisticated conditions (like matching by exec path)
